SparseGrids/tsgCacheLagrange.hpp

```diff
@@ -139,100 +139,56 @@ class CacheLagrangeDerivative : public CacheLagrange<T> {
      * - \b holds the coordinates of the canonical point to cache
      */
     CacheLagrangeDerivative(int num_dimensions, const std::vector<int> &max_levels, const OneDimensionalWrapper &rule, const double x[]) :
-            CacheLagrange<T>(num_dimensions, max_levels, rule, x) {
-        cacheDerivatives.resize(num_dimensions);
+            cache(std::vector<std::vector<T>>(num_dimensions)), offsets(rule.getPointsCount()) {
         for(int dim=0; dim<num_dimensions; dim++){
-            cacheDerivatives[dim].resize(offsets[max_levels[dim] + 1]);
+            cache[dim].resize(offsets[max_levels[dim] + 1]);
```

Comment:
* initialize that one too

Reply:
Do you mean I should initialize `cache` as follows?
```
cache(std::vector<std::vector<T>>(num_dimensions, std::vector<T>()))
```
Also, should I also apply the same changes to the `CacheLagrange` class?

Reply:
Yes, I missed the `CacheLagrange` class back when I was coding it.

Inline initialization is better than initialize and then reinitialize, whenever reasonable and possible.

Reply:
Okay. I finished making the suggested changes.